arch/arc/include/swap_macros.h

```diff
@@ -86,9 +89,14 @@
 	lr r13, [_ARC_V2_FPU_DPFP2H]
 	st_s r13, [sp, ___callee_saved_stack_t_dpfp2h_OFFSET]
 #endif
-1 :
 #endif
-
+fpu_skip_save :
+#ifdef CONFIG_ARC_DSP_SHARING
+	_save_dsp_regs
+#endif
+#ifdef CONFIG_AGU_SHARING
+	_save_agu_regs
```

Comment:
AGU is a part of DSP - so let's call `_save_agu_regs` from `_save_dsp_regs` macro in `arch/arc/core/dsp/swap_dsp_macros.h` directly.